bundle/src/test/java/com/adobe/acs/commons/packaging/impl/AssetPackagerServletImplTest.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2019 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package com.adobe.acs.commons.packaging.impl;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.jcr.Session;
+
+import com.day.cq.wcm.api.NameConstants;
+
+import org.apache.jackrabbit.vault.packaging.Packaging;
+import org.apache.jackrabbit.vault.packaging.impl.JcrPackageManagerImpl;
+import org.apache.sling.testing.mock.sling.ResourceResolverType;
+import org.apache.sling.testing.mock.sling.servlet.MockSlingHttpServletRequest;
+import org.apache.sling.testing.mock.sling.servlet.MockSlingHttpServletResponse;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import io.wcm.testing.mock.aem.junit.AemContext;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AssetPackagerServletImplTest {
+
+    @Rule
+    public final AemContext context = new AemContext(ResourceResolverType.JCR_OAK);
+
+    @Mock
+    Packaging packaging;
+
+    @InjectMocks
+    @Spy
+    final AssetPackagerServletImpl assetPackagerServlet = new AssetPackagerServletImpl();
+
+    @InjectMocks
```

Comment:
Im a bit confused by this `@InjectMocks` mostly b/c I've never used it, but docs at https://static.javadoc.io/org.mockito/mockito-core/2.27.0/org/mockito/InjectMocks.html seem to indicate it would be used to create a mock itself rather than act on an instance you've created.  That said, if I remove it the tests break, so I'm not entirely sure :D.

Reply:
So I have the `@InjectMocks` here because from what I understand it replaces a value _inside_ the object with a mocked object above. So what I'm doing is mocking the `packaging` object above and then the injection replaces the local 'packaging' object (which is null in my mocked objects but necessary to create a package) with this mocked object that has (line 134) a defined return for a package manager that lets me actually fake the package objects.

Reply:
ahhh ok, thanks for connecting the dots for me :D